puresasl/mechanisms.py

```diff
@@ -52,6 +53,8 @@ class Mechanism(object):
 
     def __init__(self, sasl, **props):
         self.sasl = sasl
+        self.qops = [QOP.AUTH]
+        self.qop = QOP.AUTH
```

Comment:
Can you clarify why this change is needed instead of using the QOPs from `sasl`?  That would be a breaking change from the current API (and we would need to remove the `qops` param from `SaslClient` if this is done).